gh-126204: Add a `sys._jit_enabled` helper (#126247)
python/cpython: Misc/NEWS.d/next/Core_and_Builtins/2024-11-10-04-59-43.gh-issue-126204.ASKDL1.rst

> Add a :func:`!sys._jit_enabled` helper to determine in runtime if JIT is enabled or not.

Reviewer: ```suggestion
Allow to determine whether the JIT is enabled at runtime via :func:`!sys._jit_enabled`.
```

If you mention this new helper, I think it's also good to add a real doc entry even though it's internal. Because this kind of change would only appear in the full changelog and could be easily forgotten. WDYT? (and if so, add a What's New entry).

It's not the first time we are documenting private utilities in `sys` (e.g., https://docs.python.org/3/library/sys.html#sys._current_frames) so I think it makes sense to document it properly.

Author: Agreed.
